apps/ui/src/composables/useActions.ts

```diff
@@ -244,12 +244,19 @@ export function useActions() {
     );
   }
 
-  async function vote(proposal: Proposal, choice: Choice, reason: string) {
-    if (!web3.value.account) return await forceLogin();
+  async function vote(
+    proposal: Proposal,
+    choice: Choice,
+    reason: string
+  ): Promise<string | null> {
+    if (!web3.value.account) {
+      await forceLogin();
+      return null;
+    }
 
     const network = getNetwork(proposal.network);
 
-    await wrapPromise(
+    const txHash = await wrapPromise(
```

Comment:
```suggestion
    return wrapPromise(
```

nit.

Reply:
Why ? this will skip the next 2 calls on addPendingVote and mixpanel 🤔

Reply:
I see, I didn't notice there are more lines because git diff collapsed those 😆 
Sorry for the confusion.